compiler/il/OMRBlock.cpp

```diff
@@ -1484,18 +1490,26 @@ OMR::Block::splitPostGRA(TR::TreeTop * startOfNewBlock, TR::CFG *cfg, bool copyE
                   convertedValueNode = TR::Node::create(convertOpCode, 1, value);
                   }
                TR::SymbolReference *storedValueSymRef = createSymRefForNode(comp, methodSymbol, convertedValueNode, self()->getExit());
-               self()->getExit()->insertBefore(TR::TreeTop::create(comp,
-                                                                     TR::Node::createWithSymRef(value,
-                                                                        comp->il.opCodeForDirectStore(convertedValDataType),
+               TR::Node *storeNode = TR::Node::createWithSymRef(value, comp->il.opCodeForDirectStore(convertedValDataType),
                                                                         1,
                                                                         convertedValueNode,
-                                                                        storedValueSymRef)));
+                                                                        storedValueSymRef);
+#ifdef J9_PROJECT_SPECIFIC
+               if (convertedValueNode->getType().isBCD())
+                  storeNode->setDecimalPrecision(convertedValueNode->getDecimalPrecision());
+#endif
+               self()->getExit()->insertBefore(TR::TreeTop::create(comp,
+                                                                     storeNode));
                TR::Node *replacementNode = TR::Node::createLoad(value, storedValueSymRef);
                if (convertedValDataType != originalValDataType)
                   {
                   TR::ILOpCodes convertOpCode = TR::ILOpCode::getProperConversion(convertedValDataType, originalValDataType, false);
                   replacementNode = TR::Node::create(convertOpCode, 1, replacementNode);
                   }
+#ifdef J9_PROJECT_SPECIFIC
+               if (convertedValueNode->getType().isBCD())
+                  replacementNode->setDecimalPrecision(convertedValueNode->getDecimalPrecision());
```

Comment:
would be nice if the Node::create or whatever did this internally in OpenJ9 for BCD typed nodes, but I won't insist on it here, but would be a good follow-up since this littering of the code with BCD type checks is not very elegant.